File: Beecrowd_Parte4/Presentes_de_Natal_2022.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
//tudo errado 
#define MAX_PRESENTES 100

// Estrutura para representar um presente
typedef struct {
    char nome[101];
    double preco;
    int preferencia;
} Presente;

// Função de comparação para qsort
int compara_presentes(const void *a, const void *b) {
    Presente *presenteA = (Presente *)a;
    Presente *presenteB = (Presente *)b;

    // Ordenar por preferência decrescente
    if (presenteA->preferencia != presenteB->preferencia) {
        return presenteB->preferencia - presenteA->preferencia;
    }
    // Ordenar por preço crescente
    else if (presenteA->preco != presenteB->preco) {
        return (presenteA->preco > presenteB->preco) ? 1 : -1;
    }
    // Ordenar por nome alfabeticamente
    else {
        return strcmp(presenteA->nome, presenteB->nome);
    }
}

int main() {
    char nomeDono[101];
    int Q;
    Presente presentes[MAX_PRESENTES];

    while (scanf("%s %d", nomeDono, &Q) == 2 && Q > 0) {
        // Leitura dos presentes
        for (int i = 0; i < Q; i++) {
            scanf("%s %lf %d", presentes[i].nome, &presentes[i].preco, &presentes[i].preferencia);
        }

        // Ordenação dos presentes
        qsort(presentes, Q, sizeof(Presente), compara_presentes);

        // Impressão conforme especificado
        printf("Lista de %s\n", nomeDono);
        for (int i = 0; i < Q; i++) {
            printf("%s - R$%.2f\n", presentes[i].nome, presentes[i].preco);
        }
        printf("\n"); // linha em branco após cada lista
    }

    return 0;
}

File: Beecrowd_Parte4/Presentes_de_Natal_2022.java
import java.util.*;
//tudo errado
class Presente implements Comparable<Presente> {
    String nome;
    double preco;
    int preferencia;

    Presente(String nome, double preco, int preferencia) {
        this.nome = nome;
        this.preco = preco;
        this.preferencia = preferencia;
    }

    @Override
    public int compareTo(Presente outro) {
        if (this.preferencia != outro.preferencia) {
            return Integer.compare(outro.preferencia, this.preferencia); // ordem decrescente de preferência
        } else if (this.preco != outro.preco) {
            return Double.compare(this.preco, outro.preco); // ordem crescente de preço
        } else {
            return this.nome.compareTo(outro.nome); // ordem alfabética do nome
        }
    }
}

public class Presentes_de_Natal_2022 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNext()) {
            String nomeDono = scanner.next();
            int Q = scanner.nextInt();
            List<Presente> presentes = new ArrayList<>();

            for (int i = 0; i < Q; i++) {
                String nomePresente = scanner.next();
                double preco = scanner.nextDouble();
                int preferencia = scanner.nextInt();
                scanner.nextLine(); // consome a quebra de linha após a preferência

                Presente presente = new Presente(nomePresente, preco, preferencia);
                presentes.add(presente);
            }

            // Ordenar os presentes de acordo com as regras especificadas
            Collections.sort(presentes);

            // Imprimir resultado conforme especificado
            System.out.println("Lista de " + nomeDono);
            for (Presente presente : presentes) {
                System.out.printf("%s - R$%.2f\n", presente.nome, presente.preco);
            }
            System.out.println(); // linha em branco após cada lista
        }

        scanner.close();
    }
}